### PR TITLE
User story 9 - Restrict Visitor Navigation 

### PR DESCRIPTION
**Pull request by MackHalliday**
- Visitor cannot visit /profile, /admin, /merchant
- Permissions add to user_controller 

- Rspec OK
- SimpleCov 100%

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
-class UsersController<ApplicationController
+class UsersController< ApplicationController
+  before_action :require_user, except: [:new, :create]
```
Comment (tschaffer1618) [inline app/controllers/users_controller.rb]: Does this level of authorization still let visitors browse the site (items and merchants) and add items to the cart?

File: spec/features/users/login_spec.rb [hunk 2]
```diff
@@ -110,9 +112,10 @@
       click_button "Submit"
 
       visit '/merchant'
+
       expect(page).to have_content("The page you were looking for doesn't exist.")
 
-      visit admin_path
+      visit '/admin'
```
Comment (tschaffer1618) [inline spec/features/users/login_spec.rb]: I like admin_path better here.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (MackHalliday) [reply app/controllers/users_controller.rb]: Yes, this only limits visitor access to the user#show/#edit/#destroy (user profile features). Visitor can still add items to cart and view merchants/items

File: spec/features/users/login_spec.rb (hunk 2 above)
Comment (MackHalliday) [reply spec/features/users/login_spec.rb]: Sorry, left over from when I was debugging. Just updated in the Github file.

**Comment by MackHalliday** [comment]
Yup, had some files with minor changes to spacing. Product of debugging and pry.

**Comment by MackHalliday** [comment]
Closing pull request - resubmitting with updates to spacing/minor formating

**Pull request closed without merging**